### ISSUE TITLE
Null Reference Exception When Deploying OpenSourceDebug to Experimental Hive With CTP 6

### ISSUE THREAD

**Issue by EricArndt** [dotnet/roslyn#1038]
![roslyn nre](https://cloud.githubusercontent.com/assets/2079517/6498977/3685ed0a-c2c1-11e4-8d6e-f8469f37dc58.PNG)

**StackTrace**: 
```
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.VsOnlineSettings.Connected(Boolean userPreviouslyOnline)
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.SessionStateMachine.OnPreStateChanged(Object sender, EventArgs e)
   at System.EventHandler.Invoke(Object sender, EventArgs e)
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.BaseUserSession.DrainEventQueue()
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.BaseUserSession.RaiseEvents(String[] propertyNames)
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.IdeUserSession.<ConnectAndInitializeSessionAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.BaseUserSession.<ConnectAsyncImpl>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.BaseUserSession.<ConnectAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.BaseUserSession.<PollConnectAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.Connected.ConnectedUser.Poller.<PollAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>b__4(Object state)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

**Comment by dpoeschl** [comment]
Hey @EricArndt, can you dig into the exception's StackTrace and add it to your bug report?

**Comment by dpoeschl** [comment]
Based on the stack trace, this bug was fixed internally (TFS #1125168). I'm looking for workarounds.

**Comment by dpoeschl** [comment]
Discussed with @EricArndt offline. We resolved the issue for him by 
    - Deleting the Roslyn hive under %localappdata%/Microsoft/VisualStudio/14.0Roslyn
    - Deleting the Roslyn hive under %appdata%/Microsoft/VisualStudio/14.0Roslyn
    - Deleting the HKCU\Software\Microsoft\VisualStudio\14.0Roslyn & 14.0Roslyn_Config folders
